app/src/main/java/com/odoo/ContactDetailActivity.java

```diff
@@ -1,13 +1,626 @@
+                        rawContactInsertIndex)
+                .withValue(Data.MIMETYPE, ContactsContract.CommonDataKinds.Email.CONTENT_ITEM_TYPE)
+                .withValue(ContactsContract.CommonDataKinds.Email.DATA, stringEmail)
+                .withValue(ContactsContract.CommonDataKinds.Email.TYPE,
+                        ContactsContract.CommonDataKinds.Email.TYPE_WORK).build());
+
+        // Website
+        ops.add(ContentProviderOperation
+                .newInsert(ContactsContract.Data.CONTENT_URI)
+                .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID,
+                        rawContactInsertIndex)
+                .withValue(Data.MIMETYPE, ContactsContract.CommonDataKinds.Website.CONTENT_ITEM_TYPE)
+                .withValue(ContactsContract.CommonDataKinds.Website.URL, stringWebsite)
+                .withValue(ContactsContract.CommonDataKinds.Website.TYPE,
+                        ContactsContract.CommonDataKinds.Website.TYPE_HOME).build());
+
+        // address, city, zip, county
+
+        address = String.valueOf(new StringBuilder(stringStreet).append(", ").append(stringStreet2));
+        if (stringStreet.equals("")) {
+            if (stringStreet2.equals("")) {
+                address = "";
+            } else {
+                address = stringStreet2;
+            }
+        } else {
+            address = stringStreet;
+        }
+
+        ops.add(ContentProviderOperation
+                .newInsert(ContactsContract.Data.CONTENT_URI)
+                .withValueBackReference(ContactsContract.Data.RAW_CONTACT_ID,
+                        rawContactInsertIndex)
+                .withValue(Data.MIMETYPE,
+                        ContactsContract.CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE)
+                .withValue(ContactsContract.CommonDataKinds.StructuredPostal.STREET, address)
+                .withValue(Data.MIMETYPE,
+                        ContactsContract.CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE)
+                .withValue(ContactsContract.CommonDataKinds.StructuredPostal.CITY,
+                        stringCity)
+                .withValue(Data.MIMETYPE,
+                        ContactsContract.CommonDataKinds.StructuredPostal.CONTENT_ITEM_TYPE)
+                .withValue(ContactsContract.CommonDataKinds.StructuredPostal.POSTCODE,
+                        stringPincode)
+                //TODO : country name
+                /*.withValue(ContactsContract.CommonDataKinds.StructuredPostal.COUNTRY,
+                        stringCountryName)*/
+                .build());
+
+        try {
+            ContentProviderResult[] res = this.getContentResolver().applyBatch(
+                    ContactsContract.AUTHORITY, ops);
+        } catch (RemoteException | OperationApplicationException e) {
+            e.printStackTrace();
+        }
+        Toast.makeText(this, "Contact Saved", Toast.LENGTH_SHORT).show();
```

Comment:
Replace with ``SnackBar`` with quick view action for recently created contact open in Contact App.